File: oo-patterns/src/main/java/victor/training/oo/creational/builder/Customer.java
package victor.training.oo.creational.builder;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Customer {

	private String name;
	private String phone;
	private List<String> labels = new ArrayList<>();
	private Address address;
	private Date createDate;

	public static class Builder {
		private final Customer customer = new Customer();

		public Builder withName(String name) {
			customer.setName(name);
			return this;
		}

		public Builder withLabel(String label) {
			customer.getLabels().add(label);
			return this;
		}

		public Builder withAddress(Address address) {
			customer.setAddress(address);
			return this;
		}

		public Customer build() {
			return customer;
		}
	}

	public Address getAddress() {
		return address;
	}

	public void setAddress(Address address) {
		this.address = address;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getPhone() {
		return phone;
	}

	public void setPhone(String phone) {
		this.phone = phone;
	}

	public List<String> getLabels() {
		return labels;
	}

	public void setLabels(List<String> labels) {
		this.labels = labels;
	}

	public Date getCreateDate() {
		return createDate;
	}

	public void setCreateDate(Date createDate) {
		this.createDate = createDate;
	}

}

File: oo-patterns-php/victor/training/oo/creational/builder/Customer.php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: VictorRentea
 * Date: 9/18/2017
 * Time: 1:31 AM
 */

namespace training\oo\creational\builder;


class Customer
{
    /* @var String */
    private $name;

    /* @var String */
    private $phone;

    /* @var String[] */
    private $labels = array();

    /* @var Address */
    private $address;

    /* @var \DateTime */
    private $createDate;

    public function __construct()
    {
        $this->labels = [];
    }

    public function addLabel(string $label): Customer {
        $this->labels[] = $label;
        return $this;
    }

    public function getName(): String
    {
        return $this->name;
    }

    public function setName(String $name): Customer
    {
        $this->name = $name;
        return $this;
    }

    public function getPhone(): String
    {
        return $this->phone;
    }

    public function setPhone(String $phone): Customer
    {
        $this->phone = $phone;
        return $this;
    }

    public function getLabels(): array
    {
        return $this->labels;
    }

    public function setLabels(array $labels): Customer
    {
        $this->labels = $labels;
        return $this;
    }

    public function getAddress(): ?Address
    {
        return $this->address;
    }

    public function setAddress(?Address $address): Customer
    {
        $this->address = $address;
        return $this;
    }

    public function getCreateDate(): \DateTime
    {
        return $this->createDate;
    }

    public function setCreateDate(\DateTime $createDate): Customer
    {
        $this->createDate = $createDate;
        return $this;
    }

}